addons/purchase/models/product.py

```diff
@@ -46,7 +52,10 @@ def get_import_templates(self):
 
     def action_view_po(self):
         action = self.env["ir.actions.actions"]._for_xml_id("purchase.action_purchase_history")
-        action['domain'] = ['&', ('state', 'in', ['purchase', 'done']), ('product_id', 'in', self.product_variant_ids.ids)]
+        action['domain'] = ['&',
+                ('state', 'in', ['purchase', 'done']),
+                ('product_id', 'in', self.with_context(active_test=False).product_variant_ids.ids)
+        ]
```

Comment:
```suggestion
        action['domain'] = [
            ('state', 'in', ['purchase', 'done']),
            ('product_id', 'in', self.with_context(active_test=False).product_variant_ids.ids),
        ]
```
Doesn't have to be that far indented, explicit `'&'` can be left out, and better to keep a trailing comma on the last line as well.

Also, not really an option in this version yet, but starting from 17.0, we could consider replacing the last line with:
```diff
-            ('product_id', 'in', self.with_context(active_test=False).product_variant_ids.ids),
+            ('product_id', 'any', [('product_tmpl_id', 'in', self.ids)]),
```